var/spack/repos/builtin/packages/py-your/package.py

```diff
@@ -10,15 +10,13 @@ class PyYour(PythonPackage):
     """Python library to read and process pulsar data in several different formats"""
 
     homepage = "https://github.com/thepetabyteproject/your"
-
-    # pypi tarball has requirements.txt missing
-    url = "https://github.com/thepetabyteproject/your/archive/refs/tags/0.6.7.tar.gz"
+    git = "https://github.com/thepetabyteproject/your.git"
 
     maintainers("aweaver1fandm")
 
     license("GPL-3.0")
 
-    version("0.6.7", sha256="f2124a630d413621cce067805feb6b9c21c5c8938f41188bd89684968478d814")
```

Comment:
Please do keep the older version and `url` in place. When you add the `main` branch, it will get picked for new installs, but we also support users who have pinned older versions (possibly with pinned dependencies), and we don't like to pull the rug from under them without older versions going through a deprecation phase first.

Reply:
I have added the URL and version back into the package as you requested.  I also added additional constraints on the py-numpy version to hopefully ensure that if version 0.6.7 of py-your is bult, it will not grab a version of py-numpy that is too new.